R/tmiic.utils.R

```diff
@@ -0,0 +1,1343 @@
+      #
+      msg_str = list_to_str (state_order$var_names[na_in_so & are_contextual],
+                             n_max=10)
+      miic_warning ("temporal checks", "the missing number of layers have been",
+        " set to 1 for contextual variables (", msg_str, ").")
+      state_order$n_layers[na_in_so & are_contextual] = 1
+      }
+    #
+    # Contextual vars done, look if still NAs on not contextual
+    #
+    na_in_so = is.na (state_order$n_layers)
+    if ( any (na_in_so) )
+      {
+      # For non contextual vars with n_layers equal to NA:
+      # - if no other var has a n_layers, go for automatic estimate
+      # - if there is an unique n_layers, apply this value to all
+      # - if there is multiple n_layers values, stop
+      #
+      uniq_vals = unique (state_order$n_layers[(!na_in_so) & (!are_contextual)])
+      if (length (uniq_vals) == 0)
+        {
+        msg_str = list_to_str (state_order$var_names[na_in_so], n_max=10)
+        miic_warning ("temporal checks", "the missing number of layers will be ",
+          " determined from data for variables ", msg_str, ".")
+        }
+      else if (length (uniq_vals) > 1)
+        {
+        miic_error ("temporal checks",
+          "some number of layers are missing and they can not be completed",
+          " automatically as multiple values are already present.")
+        }
+      else
+        {
+        msg_str = list_to_str (state_order$var_names[na_in_so], n_max=10)
+        miic_warning ("temporal checks", "the missing number of layers will be ",
+          " set to ", uniq_vals, " for variables ", msg_str, ".")
+        state_order$n_layers[na_in_so & are_contextual] = uniq_vals
+        }
+      }
+    #
+    # Check/fix invalid values:  for contextual vars, n_layers must be 1
+    #
+    wrongs = ( ( ! is.na (state_order$n_layers) )
+             & (state_order$n_layers != 1)
+             & (state_order$is_contextual == 1) )
+    if ( any (wrongs) )
+      {
+      msg_str = list_to_str (state_order$var_names[wrongs], n_max=10)
+      if (sum (wrongs) == 1)
+        miic_warning ("temporal checks", "the variable ", msg_str, ", as",
+          " contextual, has an invalid number of layers. It will be set to 1.")
+      else
+        miic_warning ("temporal checks", "several variables (", msg_str, "), as",
+          " contextual, have an invalid number of layers. They will be set to 1.")
+      state_order$n_layers[wrongs] = 1
+      }
+    #
+    # Warning if multiple values of n_layers excluding contextual
+    #
+    uniq_vals = unique (state_order$n_layers[ (!is.na (state_order$n_layers))
+                                            & (state_order$is_contextual == 0) ])
+    if (length (uniq_vals) > 1)
+      {
+      msg_str = list_to_str (uniq_vals)
+      miic_warning ("temporal checks", "different values (", msg_str,
+        ") have be defined for the number of layers.",
+        " Such setting is experimental and not recommanded.")
```

Comment:
EN (OD) + exp?

Reply:
Such setting should be avoided unless \"specific\" reason",
        " as the result will likely not be accurate.